framework/src/media/FileOutputDataSource.cpp

```diff
@@ -102,57 +126,29 @@ bool FileOutputDataSource::isPrepare()
 	return true;
 }
 
-ssize_t FileOutputDataSource::write(unsigned char* buf, size_t size)
+ssize_t FileOutputDataSource::onStreamBufferReadable(bool isFlush)
 {
-	if (!isPrepare()) {
-		return (ssize_t)EOF;
-	}
-
-	if (!buf) {
-		meddbg("buf is nullptr, hence return EOF\n");
-		return (ssize_t)EOF;
-	}
-
-	std::shared_ptr<Encoder> encoder = getEncoder();
-
-	size_t wlen = 0;
-	while (wlen < size) {
-		if (encoder) {
-			// Push data as much as possible
-			size_t pushed = encoder->pushData(buf + wlen, size - wlen);
-			if (pushed == 0) {
-				meddbg("Can not push data! Error occurred during encoding!\n");
-				break;
-			}
-
-			wlen += pushed;
-
-			// Encode data and write to file.
-			while (1) {
-				// Reuse 'wlen' bytes free space in 'buf'.
-				// Encoded data size is usually smaller than origin PCM data size.
-				size_t ret = wlen;
-				if (!encoder->getFrame(buf, &ret)) {
-					// Need push more data
-					break;
-				}
-
-				size_t written = fwrite(buf, sizeof(unsigned char), ret, mFp);
-				medvdbg("written size: %d\n", written);
-				if (written != ret) {
-					meddbg("Can not write all!\n");
-					break;
-				}
+	size_t total = 0;
+	unsigned char buf[1024];
```

Comment:
Same as above.

Reply:
Currently, stream buffer only provide `read()`,` sizeOfData()` APIs, instead of buffer pointer.
To do this, we must provide new APIs to get/release buffer addr of data payload.